albumentations/augmentations/geometric/functional.py

```diff
@@ -597,3 +592,171 @@ def safe_rotate_enlarged_img_size(angle: float, rows: int, cols: int):
+            to ``(height, width)``, otherwise ``(height, width, nb_channels)``.
+    """
+    assert (
+        distance_maps.ndim == 3
+    ), f"Expected three-dimensional input, got {distance_maps.ndim} dimensions and shape {distance_maps.shape}."
+    height, width, nb_keypoints = distance_maps.shape
+
+    drop_if_not_found = False
+    if if_not_found_coords is None:
+        drop_if_not_found = True
+        if_not_found_x = -1
+        if_not_found_y = -1
+    elif isinstance(if_not_found_coords, (tuple, list)):
+        assert (
+            len(if_not_found_coords) == 2
+        ), f"Expected tuple/list 'if_not_found_coords' to contain exactly two entries, got {len(if_not_found_coords)}."
+        if_not_found_x = if_not_found_coords[0]
+        if_not_found_y = if_not_found_coords[1]
+    elif isinstance(if_not_found_coords, dict):
+        if_not_found_x = if_not_found_coords["x"]
+        if_not_found_y = if_not_found_coords["y"]
+    else:
+        raise Exception(
+            f"Expected if_not_found_coords to be None or tuple or list or dict, got {type(if_not_found_coords)}."
+        )
+
+    keypoints = []
+    for i in range(nb_keypoints):
+        if inverted:
+            hitidx_flat = np.argmax(distance_maps[..., i])
+        else:
+            hitidx_flat = np.argmin(distance_maps[..., i])
+        hitidx_ndim = np.unravel_index(hitidx_flat, (height, width))
+        if not inverted and threshold is not None:
+            found = distance_maps[hitidx_ndim[0], hitidx_ndim[1], i] < threshold
+        elif inverted and threshold is not None:
+            found = distance_maps[hitidx_ndim[0], hitidx_ndim[1], i] >= threshold
+        else:
+            found = True
+        if found:
+            keypoints.append((hitidx_ndim[1], hitidx_ndim[0]))
+        else:
+            if not drop_if_not_found:
+                keypoints.append((if_not_found_x, if_not_found_y))
+
+    return keypoints
+
+
+def keypoint_piecewise_affine(
+    keypoint: Sequence[float],
+    matrix: skimage.transform.PiecewiseAffineTransform,
+    h: int,
+    w: int,
+) -> Tuple[float, float, float, float]:
+    x, y, a, s = keypoint
+    dist_maps = to_distance_maps([(x, y)], h, w, True)
+    dist_maps = piecewise_affine(dist_maps, matrix, 0, "constant", 0)
+    x, y = from_distance_maps(dist_maps, True, {"x": -1, "y": -1}, 0.01)[0]
+    return x, y, a, s
+
+
+def bbox_piecewise_affine(
+    bbox: Sequence[float],
+    matrix: skimage.transform.PiecewiseAffineTransform,
+    h: int,
+    w: int,
+) -> Tuple[float, float, float, float]:
+    x1, y1, x2, y2 = denormalize_bbox(tuple(bbox), h, w)
+    keypoints = [
+        (x1, y1),
+        (x2, y1),
+        (x2, y2),
+        (x1, y2),
+    ]
+    dist_maps = to_distance_maps(keypoints, h, w, True)
+    dist_maps = piecewise_affine(dist_maps, matrix, 0, "constant", 0)
+    keypoints = from_distance_maps(dist_maps, True, {"x": -1, "y": -1}, 0.01)
```

Comment:
Could you please clarify why `0.01` is used as the value for `threshold`? Does `0.01` have some special meaning?

Reply:
I don't know. This threshold is used by imgaug. A added this threshold as an argument to the transform.